### PR TITLE
[16.0][FIX] account_chart_update: avoid fiscal position duplication when it's archived

### PR DESCRIPTION
**Pull request by edlopen**
Hi y'all!

I have encountered an error when running the chart of accounts update and one of its fiscal positions is archived, an attempt is made to recreate it and an error occurs as the unique key is duplicated. This simple correction makes the wizard take into account the fiscal positions that are archived.

@moduon MT-8097

@yajo @rafaelbn @fcvalgar please review, if you can.

### CODE REVIEW

File: account_chart_update/tests/test_account_chart_update.py [hunk 1]
```diff
@@ -466,3 +466,14 @@ def test_00_matching(self):
...
+    def test_01_archived_fiscal_position(self):
+        # Test wizard won't duplicate existing fiscal positions when archived
+        fiscal_position = self.env["account.fiscal.position"].search(
+            [("name", "=", self.fp_template.name), ("company_id", "=", self.company.id)]
+        )
+        fiscal_position.active = False
+        wizard = self.wizard_obj.create(self.wizard_vals)
+        wizard.action_find_records()
+        wizard.action_update_records()
+        wizard.unlink()
```
Comment (yajo) [inline account_chart_update/tests/test_account_chart_update.py:491]: issue: just make a final assertion here, for test completeness:

```suggestion
        wizard.unlink()
        self.assertTrue(fiscal_position.exists())
        self.assertFalse(fiscal_position.active)
```

File: account_chart_update/tests/test_account_chart_update.py [hunk 2]
```diff
@@ -466,3 +466,14 @@ def test_00_matching(self):
         self.assertTrue(list(self.fp.get_external_id().values())[0])
         self.assertEqual(fp_rec.get_external_id().get(fp_id), expected_xmlid)
         wizard.unlink()
+
+    def test_01_archived_fiscal_position(self):
+        # Test wizard won't duplicate existing fiscal positions when archived
+        fiscal_position = self.env["account.fiscal.position"].search(
+            [("name", "=", self.fp_template.name), ("company_id", "=", self.company.id)]
+        )
```
Comment (yajo) [inline account_chart_update/tests/test_account_chart_update.py:476]: issue: the test is incomplete AFAICS.

The use case we're testing involves a fiscal position template with tax mappings, and a fiscal position with the same tax mappings.

The issue was when those tax mappings were creating duplicate keys, so without tax mappings, the test is incomplete.

**Review by yajo — CHANGES_REQUESTED** [review]

File: account_chart_update/tests/test_account_chart_update.py (hunk 2 above)
Comment (edlopen) [reply account_chart_update/tests/test_account_chart_update.py:476]: Ok, I've made one mapping but I'm not sure if its the right way to declare them. I'll leave this review open to check this out.

Comment (yajo) [reply account_chart_update/tests/test_account_chart_update.py:476]: If this assertion passes, then the test is OK:

```suggestion
        fiscal_position = self.env["account.fiscal.position"].search(
            [("name", "=", self.fp_template.name), ("company_id", "=", self.company.id)]
        )
        self.assertEqual(fiscal_position.tax_ids.tax_src_id.name, self.tax_template.name)
        self.assertEqual(fiscal_position.tax_ids.tax_dest_id.name, self.tax_template.name)
```

File: account_chart_update/tests/test_account_chart_update.py [hunk 3]
```diff
@@ -466,3 +466,18 @@ def test_00_matching(self):
...
+        fiscal_position.active = False
+        wizard = self.wizard_obj.create(self.wizard_vals)
+        wizard.action_find_records()
+        wizard.action_update_records()
+        wizard.unlink()
+        self.assertTrue(fiscal_position.exists())
+        self.assertFalse(fiscal_position.active)
```
Comment (yajo) [inline account_chart_update/tests/test_account_chart_update.py]: Please add again the assertion that checks tax mappings are OK.

**Review by yajo — CHANGES_REQUESTED** [review]

File: account_chart_update/tests/test_account_chart_update.py (hunk 3 above)
Comment (edlopen) [reply account_chart_update/tests/test_account_chart_update.py]: Waiting for all checks to be passed but I guess it's ready to be merged now.

Comment (yajo) [reply account_chart_update/tests/test_account_chart_update.py]: The test is failing:

```
2024-11-21 08:11:35,717 926 ERROR odoo odoo.addons.account_chart_update.tests.test_account_chart_update: FAIL: TestAccountChartUpdate.test_01_archived_fiscal_position
Traceback (most recent call last):
  File "/__w/account-financial-tools/account-financial-tools/account_chart_update/tests/test_account_chart_update.py", line 483, in test_01_archived_fiscal_position
    self.assertFalse(fiscal_position.active)
AssertionError: True is not false

Comment (edlopen) [reply account_chart_update/tests/test_account_chart_update.py]: Oh! I see. I just found out that as the account.fiscal.position.template doesn't have the 'active' field, so when the wizard tries to update it makes it active again as is the default value for such field.

So now the error has been fixed but I would raise this question: Is it desirable to make those fiscal fiscal positions active again each time you run the wizard?

Comment (yajo) [reply account_chart_update/tests/test_account_chart_update.py]: > So now the error has been fixed but I would raise this question: Is it desirable to make those fiscal fiscal positions active again each time you run the wizard?

No, disabled FPs should remain disabled.

**Review by yajo — APPROVED** [review]
/ocabot merge patch

**Review by rafaelbn — APPROVED** [review]
/ocabot merge patch

**Pull request merged**